paddle/operators/listen_and_serv_op.cc

```diff
@@ -141,10 +142,12 @@ class ListenAndServOp : public framework::OperatorBase {
             PADDLE_THROW("Can not find server side var");
           }
           detail::DeserializeFromMessage(v.second, dev_ctx, var);
+          if (var->IsType<framework::SelectedRows>()) {
+            sparse_vars.push_back(var);
```

Comment:
Do we need to reset `sparse_vars` somewhere, or stop increase its size under some condition? Otherwise it looks like the `sparse_vars` size will keep increasing.

Reply:
Sorry, I think it's a bug, I will fix this.

Reply:
`sparse_vars ` would a new one before start to receive vars for each mini-batch, but you're right, I think we could move `sparse_vars` for this function block and reset it after each mini-batch.

Reply:
Done.